### ISSUE TITLE
[Build] Fails to build on linux (BB #623)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#623]
This issue was migrated from bitbucket.
**Original Reporter:** silvermcd123
**Original Date:** 02.08.2014 18:44:42 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** closed
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623

<hr>

[  0%] Generating jemallocs_internals
/bin/sh: cmake: command not found
make[2]: **\* [dep/jemalloc/jemallocs_internals] Error 127
make[1]: **\* [dep/jemalloc/CMakeFiles/jemalloc_internals.dir/all] Error 2
make: **\* [all] Error 2

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 02.08.2014 19:02:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11572937

<hr>

i was able to fix the one up.

but this one is new:

In file included from /home/server/Servidores/wowpt/Sources/oregoncore/src/oregoncore/Master.cpp:46:
/home/server/Servidores/wowpt/Sources/oregoncore/src/shared/Debugging/UnixDebugger.h:29:22: error: demangle.h: No such file or directory
make[2]: **\* [src/oregoncore/CMakeFiles/oregon-core.dir/Master.cpp.o] Error 1
make[1]: **\* [src/oregoncore/CMakeFiles/oregon-core.dir/all] Error 2
make: **\* [all] Error 2

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 02.08.2014 21:08:13 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11573684

<hr>

I've got multiple people on IRC that use Linux and OregonCore together. Maybe jump on and let us know which version and what make you are running :)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 02.08.2014 21:56:20 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11574024

<hr>

Centos 6 and i've compiled other cores here like TrinityCore, MistCore....

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 02.08.2014 22:04:36 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11574070

<hr>

Could you search your filesystem for the demangle.h file ? Like locate demangle.h, etc... you should have that because you were able to run make which means that you have binutils, the crucial dep for unixdebugger.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 04.08.2014 13:10:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11594303

<hr>

yes and i built other cores before. i dont know why this is happening

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 04.08.2014 13:26:24 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11594672

<hr>

So ? Where is it located ?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 06.08.2014 03:34:21 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11628926

<hr>

As far as I'm aware we've had multiple builds in the past day alone with linux. This kind of stuff needs to be dealt with on the forums or in IRC. Requesting closure.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 06.08.2014 11:30:58 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11636410

<hr>

What happens when you apply this patch with `hg import` and try to recompile:

```
# HG changeset patch
# User Zaffy <zaffy@lunar.sk>
# Date 1407324564 -7200
#      Wed Aug 06 13:29:24 2014 +0200
# Node ID 3bcd012fdd20c9ce01e128b061cbb08e6a22e491
# Parent  70f1c389e4f17b8bc38917f732678e0a9fc97f4b
Fix compile for CentOS Linux

diff -r 70f1c389e4f1 -r 3bcd012fdd20 src/shared/Debugging/UnixDebugger.h
--- a/src/shared/Debugging/UnixDebugger.h   Tue Aug 05 23:36:19 2014 +0200
+++ b/src/shared/Debugging/UnixDebugger.h   Wed Aug 06 13:29:24 2014 +0200
@@ -26,9 +26,11 @@
 #define HAVE_DECL_BASENAME 1

 #include <libiberty.h>
-#include <demangle.h>
 #include <bfd.h>

+extern char *
+cplus_demangle (const char *mangled, int options);
+
 namespace UnixDebugger
 {
     void RegisterDeadlySignalHandler();
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 06.08.2014 14:32:13 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11640100

<hr>

yeah that patch solved the problem

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 06.08.2014 14:32:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11640107

<hr>

solved

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 06.08.2014 14:45:52 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/623/#comment-11640448

<hr>

Fix compile for libiberty distributions without demangle.h (linux) - closes #623

→ <<cset ce26bc5f5d61>>
